test/common/run-tests

```diff
@@ -318,12 +336,13 @@ def detect_tests(test_files, image, opts):
                     continue
                 nd = getattr(test_method, "_testlib__non_destructive", False)
                 rwa = getattr(test_method, "_testlib__retry_when_affected", True)
+                todo = getattr(test_method, "_testlib_todo", None)
```

Comment:
While rebasing #18039 to this, I noticed that the decorator does not work against a class, just against an individual `def test_*` method. That's why I check both the class and the method [here](https://github.com/cockpit-project/cockpit/pull/18039/commits/f95244f06a1579c94d54fe19d8a6751733f439ac#diff-fdfce066f8a78233d7423f73a8ab6f0e060584f8623b8558f0b2a0f131b79319R1588). It's not such a big deal, I can rework the PR to add `@todoPybridge` to all individual methods, but if it's not too much hassle, it'd be nice.

Reply:
Ah, this works for `@nondestructive` because of that `set_obj_attribute()` helper. The `no_retry_when_changed()` decorator iterates over all functions. Consistency R US!

I did a quick check that this works:
```diff
--- test/common/run-tests
+++ test/common/run-tests
@@ -336,7 +336,8 @@ def detect_tests(test_files, image, opts):
                     continue
                 nd = getattr(test_method, "_testlib__non_destructive", False)
                 rwa = getattr(test_method, "_testlib__retry_when_affected", True)
-                todo = getattr(test_method, "_testlib_todo", None)
+                todo = getattr(test_method, "_testlib_todo",
+                               getattr(test.__class__, "_testlib_todo", None))
                 if getattr(test.__class__, "provision", None):
                     # each additionally provisioned VM costs parallel test capacity
                     cost = len(test.__class__.provision)
--- test/common/testlib.py
+++ test/common/testlib.py
@@ -2213,12 +2206,12 @@ def todo(reason='', flaky=False):
     unexpected pass is considered to be an error in that case.  If flaky=True
     then a pass is not considered to be an error.
     """
-    def wrapper(func):
-        func._testlib_todo = (reason, flaky)
-        return func
+    def wrapper(testEntity):
+        testEntity._testlib_todo = (reason, flaky)
+        return testEntity
     return wrapper
 
```
(the second part is just renaming to clarify it could be a class or method).

Reply:
I added this fix as a [hack commit](https://github.com/cockpit-project/cockpit/pull/18039/commits/758293affd78e4c4a5cedb26755696159f7f3d6d) to #18039.

Reply:
... and applied here in force-push. I made an entry to my todo list to clean up the "method vs. class" approaches for our decorators, this is too wild.